README.md

```diff
@@ -1,2 +1,63 @@
 # git-files-blame
-A git command implemented with JGit that blames multiple files simultaneously
+A git command implemented with JGit that blames multiple files simultaneously.
+
+## Usage
+
+
+```java
+BlameResult result = new RepositoryBlameCommand(repo) //JGit repository to be used for the blame
+  .setFilePaths(List.of("fileA.java", "fileB.java")) //list of files to blame, all files of repository if not specified 
+  .setMultithreading(true)
+  .setStartCommit(commitId) //JGit commit id from where the blame will start, HEAD if not specified
+  .call();
+```
+
+## Our idea
+
+### Git data storage
+See details about git’s data model here, particularly about the tree of objects:
+https://git-scm.com/book/en/v2/Git-Internals-Git-Objects#_tree_objects
+
+![Tree objects](./assets/tree_graph.png)  
+
+Each commit is linked to a tree of files. The tree follows the file system structure.  
+Each node represents a file or a directory, and contains its path and the sha1 of its contents.   
+Files that are unmodified between commits (both the path and the contents are the same) will have the same sha1, pointing to the same storage. Subtrees (directories) that are unmodified will also have the same hash.
+
+Use ‘git ls-files’ to explore file trees in commits.
```

Comment:
I actually think we may be going too much into detail. IMO the goal shouldn't be to explain how git works, just what's our approach

Reply:
ok agreed, I'll keep only the "Proposal" section